website/content/en/blog/2024-06-idle-power-matters-kepler-metrics-for-public-cloud-energy-efficiency.md

```diff
@@ -0,0 +1,160 @@
+
+- Situation I - two clients requested and used the same amount of resource
+
+> In the first situation, both Client A and Client B use 10% of the node's CPU (so that each client uses 50% of the total utilization) and the total power consumption of the node is 360W. Since both clients use and request the same amount of resources, any split of energy consumption will be fair here. Then, using the fraction of resource consumption of each client (50%) and multiplying by the absolute energy consumption, 180W is associated with each client.
+
+-  Situation II - one client was evicted
+
+> In the second situation, Client B was evicted, leaving Client A alone on the server. Here, the total node power is 280W, which is entirely allocated to Client A. This power is based on the power model, where the idle power is 200W. As shown in the first situation, the dynamic power of two applications using 20% of the node CPU was 160W, which means each client had a dynamic power consumption of 80W. Therefore, since Client A is still consuming 10% of the CPU in the second situation, its dynamic power consumption remains at 80W.
+
+- Situation III - new client with higher resource utilization
+
+> Now consider a third situation with uneven resource utilization: Client A still uses 10%, but Client B now uses 40% of the node's CPU. If we only use the node's absolute power consumption of 600W, Client B's resource utilization rate of 80% results in a power consumption of 480W and 120W for Client A. However, If we isolate the idle power of 200W, Client B's power consumption will be 420W (320W for dynamic + 100W for idle) and Client A's power consumption will be 180W (80W for dynamic as it is still using 10% of the CPU + 100W for idle).
+
+- Situation IV - new client with lower resource utilization
+
+> The situation IV is similar to the third situation, but Client B is using fewer resources than Client A, further illustrating the potential for unfair distribution of idle power.
+
+As we can see In the third scenario, isolating the idle power makes the results more predictable and easier to understand. This approach fairly divides the constant power (which does not vary with resource consumption) and defines the dynamic power entirely based on resource utilization. In more details, using the power model, we know that Client A's dynamic power is 80W. If we allocate idle power only using the resource ration and absolute power, Client A receives only 40W of idle power, leaving Client B with 160W of idle power.
+
+These unequal distributions of idle power highlights the unfairness, as idle power consumption is not related to resource utilization. On the other hand, applying the power consumption ratio to resource request does not reflect the real dynamic power consumption according to the real resource utilization. 
```

Comment:
```suggestion
These unequal distributions of idle power highlight unfairness because idle power consumption isn't tied to resource utilization. Applying the power consumption ratio based on resource requests doesn't accurately reflect actual dynamic power consumption corresponding to real resource utilization.
```